rpc-client/src/main.rs

```diff
@@ -102,9 +104,15 @@ impl Client {
             wallet: WalletProxy::new(client.clone()),
             validator: ValidatorProxy::new(client.clone()),
             network: NetworkProxy::new(client.clone()),
-            zkp_component: ZKPComponentProxy::new(client),
+            zkp_component: ZKPComponentProxy::new(client.clone()),
+            ws_client: client,
         })
     }
+
+    /// Closes the WS connection
+    pub async fn close(&mut self) {
+        let _ = self.ws_client.close().await;
```

Comment:
Remove clippy warning:
```suggestion
        self.ws_client.close().await;
```